### PR TITLE
Use 'required' tag instead of comments in schema

### LINKED ISSUE #82: Use JSON schemas required
The required element is now available since we migrated from protobuf, this allows us to be specific in the schema about which elements are required to ease validation.

To close this, edit the schemas to make require all elements tagged required in the comments and demonstrate some validation tests.

### LINKED ISSUE #112: checksum in the GET /dataobject and GET /databundle endpoints
In the GET /dataobject and GET /databundle endpoints checksum is a required field. Is the intended use of this parameter to filter dataobjects and databundles by their checksum values? If not then what is the intended use?

### PR DESCRIPTION
**Pull request by natanlao**
Closes #82. Closes #112.

Given #118 and #127 (i.e. all parameters to `/dataobjects` and `/databundles` are supposed to be optional) is it enough that the endpoints are already implicitly tested in the test suite? https://github.com/ga4gh/data-object-service-schemas/blob/8dc870a10320cf2c03d7cd7d9b4735688f87fd9f/python/test/test_server.py#L295

### CODE REVIEW

**Comment by natanlao** [comment]
Tests are failing because of #118 and #127. Since we don't want to break master, and they're both one-line fixes that are related to this ticket, I'll add their fixes to this PR.

**Comment by natanlao** [comment]
This also closes #112.

**Comment by david4096** [comment]
Yes, I expect that this will break some downstream code since we are now using schematized validation. These are good things to break as we strengthen the quality of the schematization.

**Comment by natanlao** [comment]
I misinterpreted the docs - I'll revise this PR.

**Comment by david4096** [comment]
Would you also please add a test that shows the server giving an expected error response when a required field isn't provided? This will be be addressed more fully in https://github.com/ga4gh/data-object-service-schemas/issues/133

**Comment by natanlao** [comment]
Ah sorry, I accidentally pushed to my fork instead of this repo. Here it is.

File: python/test/test_server.py [hunk 1]
```diff
@@ -622,3 +659,20 @@ def test_no_find(self):
...
             self.assertEqual(e.status_code, 404)
+
+    def test_schema_required(self):
+        """
+        Tests that the server properly rejects a request
+        missing a parameter that is marked as required.
+        """
+        CreateDataObjectRequest = self._models.get_model('CreateDataObjectRequest')
+        DataObject = self._models.get_model('CreateDataObjectRequest')
+        # Missing the `id` parameter
```
Comment (david4096) [inline python/test/test_server.py]: Thanks for this!

**Pull request merged**